File: dotnet/SpiderEngine/Model/SpiderTaskInfo.cs
namespace SpiderEngine.Model;

public class SpiderTaskInfo
{
    public string Name { get; set; } = default!;
    public string Url { get; set; } = default!;
    public int? Success { get; set; }
    public int Delay { get; set; }
    public SpiderMethod Method { get; set; } = default!;
    public SpiderPayload? Payload { get; set; } = default!;
    public SpiderParserInfo? Content { get; set; } = default!;
    public IEnumerable<SpiderKeyPathPair>? Header { get; set; } = default!;
}

public enum SpiderMethod
{
    Get,
    Post,
}

File: android/src/main/kotlin/com/itoken/team/wutspider/model/SpiderTaskInfo.kt
package com.itoken.team.wutspider.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SpiderTaskInfo(
    val name: String? = null,
    val url: String,
    val success: Int = 200,
    val method: SpiderMethod = SpiderMethod.GET,
    val delay: Long = 0,
    val payload: SpiderPayload? = null,
    val content: SpiderParserInfo? = null,
    val header: List<SpiderKeyPathPair>? = null
)

enum class SpiderMethod {
    @SerialName("get")
    GET,
    @SerialName("post")
    POST
}